ui/jobs/jobs.js

```diff
@@ -1563,6 +1564,95 @@ class Bars {
     };
   }
 
+  setupSch() {
+    let gcd = kUnknownGCD;
+
+    let aetherflowStackBox = this.addResourceBox({
+      classList: ['sch-color-aetherflow'],
+    });
+
+    let fairyGaugeBox = this.addResourceBox({
+      classList: ['sch-color-fairygauge'],
+    });
+
+    let bioBox = this.addProcBox({
+      id: 'sch-procs-bio',
+      fgColor: 'sch-color-bio',
+      scale: gcd,
+      threshold: gcd + 1,
+    });
+
+    let aetherflowBox = this.addProcBox({
+      id: 'sch-procs-aetherflow',
+      fgColor: 'sch-color-aetherflow',
+      scale: gcd,
+      threshold: gcd * 3,
+    });
+
+    let lucidBox = this.addProcBox({
+      id: 'sch-procs-luciddreaming',
+      fgColor: 'sch-color-lucid',
+      scale: gcd,
+      threshold: gcd + 1,
+    });
+
+    this.jobFuncs.push((jobDetail) => {
+      let aetherflow = jobDetail.aetherflowStacks;
+      let fairygauge = jobDetail.fairyGauge;
+      let milli = (jobDetail.fairyMilliseconds / 1000).toFixed(0);
+      aetherflowStackBox.innerText = aetherflow;
+      fairyGaugeBox.innerText = fairygauge;
+      let f = fairyGaugeBox.parentNode;
+      if (milli != 0) {
+        f.classList.add('bright');
+        fairyGaugeBox.innerText = milli;
+      } else {
+        f.classList.remove('bright');
+        fairyGaugeBox.innerText = fairygauge;
+      }
+
+      aetherflowBox.threshold = gcd * aetherflow;
```

Comment:
I think this should be `aetherflowBox.threshold = this.gcdSpell() + 1;`, maybe?

Reply:
Not only this box tell player to use a readied aetherflow, but also annouce he to spend his surplus aetherflow stacks. ( ･ิϖ･ิ)
Assume aetherflow stacks always be consumed as energy drain, we'd spend at least one gcd to consume one stack.
so when our thrifty scholar has 3 stacks, he must be announced when aetherflow cooldown <= 3 * gcd, make sure the profit maximization.
Also, for a confident scholar, when he has no stacks, he should only be announced when aetherflow is ready.
So I think it is not bad.

Reply:
Oops sorry, I meant `this.gcdSpell() * aetherflow + 1`.  Mostly trying to suggest using `this.gcdSpell()` instead of the default `gcd` variable?

Reply:
Oh that's reasonable. Thanks for pointing out!